x-pack/filebeat/module/salesforce/login/ingest/pipeline.yml

```diff
@@ -0,0 +1,368 @@
+---
+description: Pipeline for parsing salesforce login logs
+processors:
+- set:
+    field: event.ingested
+    value: '{{_ingest.timestamp}}'
+- remove:
+    field:
+      - message
+    ignore_missing: true
+
+- set:
+    field: salesforce.access_mode
+    value: rest
+    if: ctx?.json?.TIMESTAMP_DERIVED != null
+    ignore_failure: true
+- set:
+    field: salesforce.access_mode
+    value: stream
+    if: ctx?.json?.EventDate != null
+    ignore_failure: true
+
+- date:
+    field: json.TIMESTAMP
+    target_field: "@timestamp"
+    formats: ["yyyyMMddHHmmss.SSS"]
+    ignore_failure: true
+- date:
+    field: json.EventDate
+    target_field: "@timestamp"
+    formats:
+    - ISO8601
+    ignore_failure: true
+
+- rename:
+    field: json.API_TYPE
+    target_field: salesforce.login.api_type
+    ignore_missing: true
+- rename:
+    field: json.ApiType
+    target_field: salesforce.login.api_type
+    ignore_missing: true
+- rename:
+    field: json.AUTHENTICATION_METHOD_REFERENCE
+    target_field: salesforce.login.auth_service_id
+    ignore_missing: true
+- rename:
+    field: json.AuthMethodReference
+    target_field: salesforce.login.auth_method_reference
+    ignore_missing: true
+- rename:
+    field: json.LoginType
+    target_field: salesforce.login.login_type
+    ignore_missing: true
+- rename:
+    field: json.PolicyOutcome
+    target_field: salesforce.login.policy_outcome
+    ignore_missing: true
+- rename:
+    field: json.AuthServiceId
+    target_field: salesforce.login.auth_service_id
+    ignore_missing: true
+- rename:
+    field: json.EvaluationTime
+    target_field: salesforce.login.evaluation_time
+    ignore_missing: true
+- rename:
+    field: json.ClientVersion
+    target_field: salesforce.login.client_version
+    ignore_missing: true
+- rename:
+    field: json.LoginGeoId
+    target_field: salesforce.login.login_geo_id
+    ignore_missing: true
+- rename:
+    field: json.LoginHistoryId
+    target_field: salesforce.login.login_history_id
+    ignore_missing: true
+- rename:
+    field: json.CreatedById
+    target_field: salesforce.login.created_by_id
+    ignore_missing: true
+- rename:
+    field: json.REQUEST_STATUS
+    target_field: salesforce.login.request_status
+    ignore_missing: true
+- rename:
+    field: json.API_VERSION
+    target_field: salesforce.login.api_version
+    ignore_missing: true
+- rename:
+    field: json.ApiVersion
+    target_field: salesforce.login.api_version
+    ignore_missing: true

```

Comment:
Is there a need for two very similar renaming operations?

Reply:
`API_VERSION`: comes from httpjson (REST API)
`ApiVersion`: comes from cometd (Streaming API)
This was the case. But the filesets would be updated to `login-rest` and `login-stream` to separate them and it should not be an issue anymore.